### ISSUE TITLE
Consolidate disparate images using ENTRYPOINT

### ISSUE THREAD

**Issue by cweider** [freelawproject/courtlistener#3134]
Within [Dockerfile](https://github.com/freelawproject/courtlistener/commit/a2fbbabfd52f99791e2738360996d7c86bc51de3), there are five “leaf-images”: `celery`, `web-dev`, `web-prod`, `rss-scraper`, and `retry-webhooks`. After #3107 and #3130, it’s apparent these images are minimally differentiated. They only vary in the `CMD` instruction used:

```
FROM python-base as celery

USER www-data
CMD celery
    --app=cl worker \
    #...


FROM python-base as web-dev

USER www-data
CMD python /opt/courtlistener/manage.py migrate && \
    python /opt/courtlistener/manage.py createcachetable && \
    python /opt/courtlistener/manage.py runserver 0.0.0.0:8000


FROM python-base as web-prod

USER www-data
CMD gunicorn cl.asgi:application \
    --chdir /opt/courtlistener/ \
    #...


FROM python-base as rss-scraper

USER www-data
CMD /opt/courtlistener/manage.py scrape_rss


FROM python-base as retry-webhooks

USER www-data
CMD /opt/courtlistener/manage.py cl_retry_webhooks
```

This is valid, but it is not the only way to accomplish their purpose of making it possible create differently behaving containers from one common image (here, `python-base`). The other conventional approach is a single image containing a helper script (e.g. `docker-entrypoint.sh`) that can be used to launch the different processes depending for different types of containers. To support this pattern, Docker has the [`ENTRYPOINT`](https://docs.docker.com/engine/reference/builder/#entrypoint) instruction.

In Courtlistener’s case, `ENTRYPOINT` would be especially helpful because it would drastically simplify image management. There’d many fewer images to build, cache, tag, and push (see [`Makefile`](https://github.com/freelawproject/courtlistener/commit/722d8626ffe051ba33aa470fabb047476cf6d4d8)). Chances are that `Makefile` could be replaced with a `build-push-action` after this change.

A less recognized benefit of this change would be that the different processes run on a container’s start would be centralized within `docker-entrypoint.sh`. This makes the behavior more visible as well as accessible for testing.

Making this change requires two coordinated bits of work:
1. Shifting the behavior in `CMD` instructions to an entrypoint script. There are some [best practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#entrypoint) that would be worth reading.
2. Changing Kubernetes deploy in order to vary the commands used by different pods that currently rely on different images. The documentation on [command and arguments for a container](https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/), I think, explains how this is done.

**Comment by mlissner** [comment]
Sounds great, thank you.

**Comment by ss108** [comment]
@cweider In the `docker-build` GitHub action, there is a step to collect static assets and check whether there are any unapplied database migrations: https://github.com/freelawproject/courtlistener/blob/main/.github/workflows/docker-build.yml#L46

In addition to considering whether there is a smoother way of handling the migrations, should we also separate `collectstatic` from the migrations check? As it is right now, an error from `collectstatic` can result in the creation of a new pod, a message to Mike that leads him to think he needs to shell into that pod to migrate the db, etc. 

And as a general matter in just seems kind of wrong to treat these two separate and seemingly unrelated tasks as if they together constitute a single build step.

**Comment by mlissner** [comment]
You're right, that is kind of annoying. I was just living with it. I didn't even consider that a better way might be possible. If you do split it up, make sure you carry over the timeout flag I just added. That needs to go with whichever task happens first, because it just takes Github a while to download and start the image the first time.

**Comment by ss108** [comment]
@mlissner Thanks, sounds good.

**Comment by mlissner** [comment]
OK, after a number of tweaks, this is in place. Closing with huge appreciation to you, @ss108!
